.github/workflows/build-image-job.yaml

```diff
@@ -0,0 +1,128 @@
+name: Build docker image
```

Comment:
You can also make `.github/actions` if you want a bit of re-usable workflow code.

This being an actual workflow might be confusing to maintainers as they'd think they could run this independently whereas its meant to be run as part of a larger job of several image builds.

An action makes that clearer to me.

Reply:
I had some issues moving it to `.github/actions`, it seems like GH searches the file in the remote repo instead of the current branch. so for now I left it in the workflows dir (I did not want to spend too much time to investigate this)

Reply:
Yeah understood - I'm happy to take a look at that as a follow up. We have some familiarity with internal repos and doing this.